src-ui.v3/src/API.hs

```diff
@@ -358,3 +282,283 @@ labelModCml = uncap        . dropWhile (not . C.isUpper)
+  , _getV2PackageReportDetail :: Dynamic t (Either Text PkgN) -> Dynamic t (Either Text PkgIdxTs) -> Dynamic t (Either Text Ver) -> Dynamic t (Either Text CompilerID) -> Event t () -> m (Event t (ReqResult () CellReportDetail))
+  , _getV2PackageTags :: Dynamic t (Either Text PkgN) -> Event t () -> m (Event t (ReqResult () (Vector TagN)))
+  , _getV2Queue :: Event t () -> m (Event t (ReqResult () (Vector QEntryRow)))
+  , _getV2QueuePkg :: Dynamic t (Either Text PkgN) -> Event t () -> m (Event t (ReqResult () (Vector QEntryRow)))
+  , _putV2Queue :: Dynamic t (Either Text PkgN) -> Dynamic t (Either Text PkgIdxTs) -> Dynamic t (Either Text QEntryUpd) -> Event t () -> m (Event t (ReqResult () (QEntryRow)))
+  , _getV2TagsWithPackage :: Dynamic t (QParam Bool) -> Event t () -> m (Event t (ReqResult () (Map TagN (Vector PkgN))))
+  , _getV2TagsWithoutPackage :: Dynamic t (QParam Bool)  -> Event t () -> m (Event t (ReqResult () (Vector TagN)))
+  , _putV2PackageTags :: Dynamic t (Either Text TagN) -> Dynamic t (Either Text PkgN) -> Event t () -> m (Event t (ReqResult () NoContent))
+  , _deleteV2PackageTags :: Dynamic t (Either Text TagN) -> Dynamic t (Either Text PkgN) -> Event t () -> m (Event t (ReqResult () NoContent))  
+  , _getV2UnitInfo :: Dynamic t (Either Text UUID) -> Event t () -> m (Event t (ReqResult () UnitIdInfo))
+  , _getV2User :: Dynamic t (Either Text UserName) -> Event t () -> m (Event t (ReqResult () UserPkgs))
+  , _getV2Workers :: Event t () -> m (Event t (ReqResult () (Vector WorkerRow)))
+  , _getV2WorkersPkg :: Dynamic t (Either Text PkgN) -> Event t () -> m (Event t (ReqResult () (Vector WorkerRow)))
+  }
+makeLenses ''ClientFuns
+
+getClient :: forall t m. (SupportsServantReflex t m) 
+  => ClientFuns t m 
+getClient = mkClientFuns'' burlNew
+  where 
+    mkClientFuns'' bp = ClientFuns { .. }
+      where
+        (      _getV2IdxStates 
+          :<|> _getV2Info
+          :<|> _getV2Packages
+          :<|> _getV2PackagesHistory
+          :<|> _getV2PackageHistory
+          :<|> _getV2PackageReports
+          :<|> _getV2PackageReportSummary
+          :<|> _getV2PackageReportDetail
+          :<|> _getV2PackageTags
+          :<|> _getV2Queue
+          :<|> _getV2QueuePkg
+          :<|> _putV2Queue
+          :<|> _getV2TagsWithPackage
+          :<|> _getV2TagsWithoutPackage
+          :<|> _putV2PackageTags
+          :<|> _deleteV2PackageTags
+          :<|> _getV2UnitInfo
+          :<|> _getV2User
+          :<|> _getV2Workers
+          :<|> _getV2WorkersPkg ) = (clientWithOpts (Proxy :: Proxy API) Proxy (Proxy :: Proxy ()) (constDyn bp) tweakRequest) :: Client t m API ()
+
+type ValidationErrors = Map Text [Text]
+type ClientRes t a = (Event t a, Event t ClientError, Dynamic t Bool)
+  
+data ClientError
+  =   Forbidden
+  | NotFound
+  | Unauthorised
+  | FailedValidation (Maybe (ErrorBody ValidationErrors))
+  | OtherError Word Text
+  deriving (Show)
+  
+data ErrorBody errors = ErrorBody
```

Comment:
This probably isn't necessary(see comment on 393) or at least need to match what the matrix-ci returns in the body for errors.